dev/ProgressBar/ProgressBar.cpp

```diff
@@ -93,32 +94,47 @@ void ProgressBar::UpdateStates()
     }
     else if (!IsIndeterminate())
     {
-        SetProgressBarIndicatorWidth();
         winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
     }
 }
 
 void ProgressBar::SetProgressBarIndicatorWidth()
 {
+    const auto templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+
     if (auto&& progressBar = m_layoutRoot.get())
     {
         if (auto&& progressBarIndicator = m_progressBarIndicator.get())
         {
             const double progressBarWidth = progressBar.ActualWidth();
+            const double prevIndicatorWidth = progressBarIndicator.ActualWidth();
             const double maximum = Maximum();
             const double minimum = Minimum();
             const auto padding = Padding();
 
-            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            m_isUpdating = true; // Adds "Updating" state in between to trigger RepositionThemeAnimation Visual Transition in ProgressBar.xaml when reverting back to previous state
+            UpdateStates();
+
+            if (IsIndeterminate())
+            {
+                progressBarIndicator.Width(progressBarWidth * 0.4);
+            }
```

Comment:
Does this fix the bug where changing the value while the indeterminate animation was playing changed the indicator's size?

Reply:
Yes. It also makes more sense to put this line in the `SetProgressBarIndicatorWidth()` function.